content/blog/2016/2016-05-18-external-workspace-manager-plugin.adoc

```diff
@@ -0,0 +1,140 @@
+I am following a major in Software Engineering at the Babes-Bolyai University of Cluj-Napoca, Romania.
+I have more than two years hands-on experience working in Software Development.
+I am always learning, always looking for new challenges.
+I want to start contributing to the open source community and this is a stating point for me.
+
+=== Project summary
+Currently, Jenkins’ build workspace may become very large in size due to the fact that some compilers generate
+very large volumes of data.
+The existing plugins that share the workspace across builds are able to do this by copying the files from
+one workspace to another, process which is inefficient.
+A solution is to have a Jenkins plugin that is able to manage and reuse the same workspace between multiple builds.
+
+As part of the link:https://summerofcode.withgoogle.com/[Google Summer of Code] 2016 I will be working on
+the __External Workspace Manager__ plugin.
+My mentors for this project are link:https://github.com/oleg-nenashev[Oleg Nenashev]
+and link:https://github.com/martinda[Martin d'Anjou].
+This plugin aims to provide an external workspace management system.
+It should facilitate workspace share and reuse across multiple Jenkins jobs.
+It should eliminate the need to copy, archive or move files.
+The plugin will be written for link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Plugin[Pipeline] jobs.
+
+=== Usage
+
+Let's assume that we have two Jenkins jobs, one called `upstream` and the other one called `downstream`.
+In the upstream job, we clone the repository and build the project, and in the downstream job we run the tests.
+In the downstream job we don't want to clone and re-build the project, we need to use the same
+workspace created in the upstream job.
+We have to be able to do so without copying the workspace from one location to another.
+
+===== Prerequisites
+1. Multiple physical disks accessible to Master.
+2. The same physical disks must be accessible to Nodes.
+3. In the Jenkins global configuration, define a disk pool (or many) that will contain the physical disks.
+4. In each Node configuration, define the mounting point from the current node to each physical disk.
+
+===== Pipeline
+Allocate the external workspace from a disk pool defined in the Jenkins global configuration.
+
+[source,groovy]
+----
+// upstream job
+def extWorkspace = exwsAllocate id: 'diskpool1'
+----
+
+This step will allocate a directory on a selected physical disk from the `diskpool1`.
+The resulting workspace path will be: `/physicalPathOnDisk/$JOB_NAME/$BUILD_NUMBER`.
+
+Assume that the selected disk is `disk1` having defined the physical path as: `/jenkins-project/disk1/`.
+The `$JOB_NAME` is __upstream__ and the `$BUILD_NUMBER` is __14__, so the resulting path is:
+`/jenkins-project/disk1/upstream/14`.
+
+Execute the `exws` step within a `node` and apply the object returned by the `exwsAllocate` step.
+
+[source,groovy]
+----
+// upstream job
+node ('node1') {
```

Comment:
The reader may not be a user of the `stage()` step, so I would mention that the `stage` step is optional.

Also, it could be two different Pipeline jobs reusing the same workspace.